packages/twenty-server/src/database/commands/upgrade-version-command/upgrade.command.ts

```diff
@@ -23,12 +28,90 @@ import { CopyTypeormMigrationsCommand } from 'src/database/commands/upgrade-vers
 import { MigrateWorkflowEventListenersToAutomatedTriggersCommand } from 'src/database/commands/upgrade-version-command/0-53/0-53-migrate-workflow-event-listeners-to-automated-triggers.command';
 import { RemoveRelationForeignKeyFieldMetadataCommand } from 'src/database/commands/upgrade-version-command/0-53/0-53-remove-relation-foreign-key-field-metadata.command';
 import { UpgradeSearchVectorOnPersonEntityCommand } from 'src/database/commands/upgrade-version-command/0-53/0-53-upgrade-search-vector-on-person-entity.command';
+import { FixCreatedByDefaultValueCommand } from 'src/database/commands/upgrade-version-command/0-54/0-54-created-by-default-value.command';
 import { FixStandardSelectFieldsPositionCommand } from 'src/database/commands/upgrade-version-command/0-54/0-54-fix-standard-select-fields-position.command';
 import { TwentyConfigService } from 'src/engine/core-modules/twenty-config/twenty-config.service';
 import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
 import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
 import { SyncWorkspaceMetadataCommand } from 'src/engine/workspace-manager/workspace-sync-metadata/commands/sync-workspace-metadata.command';
-import { FixCreatedByDefaultValueCommand } from 'src/database/commands/upgrade-version-command/0-54/0-54-created-by-default-value.command';
+
+const execPromise = promisify(exec);
+
+@Injectable()
+export class DatabaseMigrationService {
+  private logger = new Logger(DatabaseMigrationService.name);
+
+  constructor(
+    @InjectRepository(Workspace, 'core')
+    private readonly workspaceRepository: Repository<Workspace>,
+  ) {}
+
+  async shouldRunMigrations(): Promise<boolean> {
+    try {
+      // Check if any workspace exists
+      const workspaceCount = await this.workspaceRepository.count();
+
+      if (workspaceCount === 0) {
+        // If no workspaces, run migrations (fresh installation)
+        this.logger.log(
+          'No workspaces found. Running migrations for fresh installation.',
+        );
+
+        return true;
+      }
+
+      const version053Count = await this.workspaceRepository.count({
+        where: {
+          version: Like('0.53%'),
+        },
+      });
+
+      if (version053Count > 0) {
+        this.logger.log(
+          `Found ${version053Count} workspace(s) on version 0.53. Will run database migrations.`,
+        );
+
+        return true;
+      } else {
+        this.logger.log(
+          'No workspace found on version 0.53. Skipping database migrations.',
+        );
+
+        return false;
+      }
+    } catch (error) {
+      this.logger.error('Error checking workspace versions:', error);
```

Comment:
we should catch the error. If the error looks like we have no db, we should return true, otherwise we should stop